lib/fluent/engine.rb

```diff
@@ -19,6 +19,16 @@ module Fluent
   require 'fluent/root_agent'
 
   class EngineClass
+    class DummyMessagePackFactory
+      def packer(io = nil)
+        MessagePack::Packer.new(io)
+      end
+
+      def unpacker(io = nil)
+        MessagePack::Unpacker.new(io)
```

Comment:
I think packer and unpacker methods should be:

```
def packer(*args)
  MessagePack::Packer.new(*args)
end
```

because Packer and Unpacker accepts some options such as `allow_unknown_ext: true` in addition to io.